content/posts/2019-05-21---Working-In-Ember.md

```diff
@@ -0,0 +1,174 @@
+
+I don't think I would have picked Ember on my own.
+I remember during a job interview, the interviewer had asked me what brought me to Ember.
+My response was something about the community and the conventions and the same things I've heard other people say 100 times.
+The truth is: I wrote Ember because that's what my company was doing.
+And I think that's a completely valid response.
+We can list all the reasons about the time Ember can save and the ergonomics, but not everyone gets to pick their company's tech stack (in fact, _most_ of us don't), regardless of our arguments.
+
+Obviously, I really like Ember and I would have liked a job where I could keep writing it.
+Unfortunately, that just isn't that easy because most companies just aren't writing Ember.
+React has almost become a default in the web dev community and regardless of whether you *like* it or not, the truth is that you are more employable if you know it.
+
+So, I started volunteering some time at a pre-funding startup with a mission I believed in to get some experience in React (since I wasn't going to get it at work).
+That's when I learned how much easier it is to get started with a library that just cares about the view layer.
+
+Secondly, I formed a lot of good habits because of Ember in terms of structuring an application, writing tests, handling how data moves through my components and more.
+But I also formed some bad habits without realizing it.
+I didn't quite know how to write native JavaScript classes.
+Things like `Component.extend({})` over `Foo extends Component`, `init` over `constructor`, `this._super` or `super()` and more, made me feel like I had fallen behind other JavaScript developers.
+
+<figure>
+	<blockquote>
+		<p>But Ember paved the way! Those things weren't around when Ember started!</p>
+    <footer>
+			<cite>— People who miss the point.</cite>
+		</footer>
+	</blockquote>
+</figure>
+
+I get that Ember helped pave the way for these things because weren't available yet.
+I also get that "stability without stagnation" means that sweeping changes can't be implemented as the goal is to not break every app out there with each release.
+But it still boils down to me not knowing something that the rest of the Web Dev world has figured out.
+Take [`broccoli.js` for example](https://github.com/broccolijs/broccoli).
+I learned a lot of about trees in broccoli, build pipelines and more!
+But everyone else in the world was using [webpack](https://webpack.js.org/) and I knew absolutely nothing about it.
+My skills didn't feel transferable.
+
+<figure>
+	<blockquote>
+		<p>Just because you can explain why something is bad, doesn't mean it's not bad</p>
+    <footer>
+			<cite>— Me.</cite>
+		</footer>
+	</blockquote>
+</figure>
+
+Even if your current company writes Ember and you love your job and you never want to leave, there may come a time when that isn't the case.
+Wouldn't you want to feel like you could find another job quickly if you had to?
```

Comment:
There is a case to be made for being a better developer in any framework from learning more than one.